### PR TITLE
Fixed IVsPackageInstallerService APIs to return correct install path

### PR DESCRIPTION
**Pull request by jainaashish**
Fixes internal bug# 482522

Issue is, we started supporting `RestorePackagesPath` in VS to have custom packages path for `PackageReference` based projects but we never updated `IVsPackageInstallerService` APIs to start consuming this property to return correct install path for the packages.

@rrelyea

### CODE REVIEW

File: src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstallerServices.cs [hunk 1]
```diff
@@ -101,6 +104,18 @@ public IEnumerable<IVsPackageMetadata> GetInstalledPackages()
                 });
         }
 
+        private async Task<FallbackPackagePathResolver> GetPackagesPathResolverAsync(BuildIntegratedNuGetProject project)
+        {
+            // To get packagesPath for build integrated projects, first read the packageSpec to know if
+            // RestorePackagesPath property was specified. If yes, then use that property to get packages path
+            // otherwise use global user cache folder from _settings.
+            var context = new DependencyGraphCacheContext();
+            var packageSpec = (await project.GetPackageSpecsAsync(context)).First();
```
Comment (emgarten) [inline src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstallerServices.cs]: this should be fine, but you should take the first spec that is of type PackageReference and not a tool spec. 
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.ProjectModel/ProjectStyle.cs#L21

File: src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstallerServices.cs [hunk 2]
```diff
@@ -101,6 +104,18 @@ public IEnumerable<IVsPackageMetadata> GetInstalledPackages()
...
+            var packagesPath = VSRestoreSettingsUtilities.GetPackagesPath(_settings, packageSpec);
+
+            return new FallbackPackagePathResolver(packagesPath, SettingsUtility.GetFallbackPackageFolders(_settings));
```
Comment (emgarten) [inline src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstallerServices.cs]: This needs to be VSRestoreSettingsUtilities.GetFallbackFolders

SettingsUtility will have the same issue for fallback folders as it did for the global packages folder.

**Review by emgarten — APPROVED** [review]

**Review by emgarten — CHANGES_REQUESTED** [review]
this needs to read fallback folders in the same way

File: src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstallerServices.cs (hunk 2 above)
Comment (jainaashish) [reply src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstallerServices.cs]: yeah, makes sense!

**Review by emgarten — APPROVED** [review]
LGTM 🚀

**Pull request merged**